base/complex.jl

```diff
@@ -424,7 +424,8 @@ function log2(z::Complex)
     a/log(oftype(real(a),2))
 end
 
-function exp(z::Complex)
+exp(z::Complex) = exp(float(z))
```

Comment:
Why can't type inference figure this out given `isnan(::Int)` and `isfinite(::Int)` are both constants? I was thinking about doing this a while ago (before we had constant propagation in type inference) but this adds branches (`isfinite` one in particular) that seems hard for LLVM to eliminat